packages/build/src/plugins_core/functions/index.ts

```diff
@@ -19,6 +20,45 @@ const getBundlers = (results: Awaited<ReturnType<typeof zipFunctions>> = []) =>
       .filter(Boolean) as NodeBundlerName[],
   )
 
+// see https://docs.netlify.com/functions/trigger-on-events/#available-triggers
+const eventTriggeredFunctions = new Set([
+  'deploy-building',
+  'deploy-succeeded',
+  'deploy-failed',
+  'deploy-deleted',
+  'deploy-locked',
+  'deploy-unlocked',
+  'submission-created',
+  'split-test-activated',
+  'split-test-deactivated',
+  'split-test-modified',
+  'identity-validate',
+  'identity-signup',
+  'identity-login',
+])
+
+const validateCustomRoutes = async function (zisiResults: Awaited<ReturnType<typeof zipFunctions>>) {
+  for (const { routes, name, schedule } of zisiResults) {
+    if (!routes || routes.length === 0) continue
+
+    if (schedule) {
+      const error = new Error(
+        `Scheduled functions must not specify a custom route. Please remove the "routes" configuration or pick a different name for the function. Learn more about scheduled functions at https://docs.netlify.com/functions/scheduled-functions.`,
```

Comment:
nit: I would use a `ntl.fyi` short link here. It's more consistent with with other messaging and lets us point it to somewhere else if needed.

Reply:
good idea, done in https://github.com/netlify/build/pull/5344/commits/9fd107b4bcbe03f3f8775537f006706ea68c5e42